e2e/async-initialization.spec.js

```diff
@@ -0,0 +1,31 @@
+const test = require('ava');
+
+const WorkerNodes = require('../');
+const { fixture } = require('./utils');
+
+test('should not mark worker as ready until module fully initialized', t => {
```

Comment:
should we create a test for failing async initialization? maybe we should have some kind of timeout? what happens, when intialization never emits ready event? I vote for adding initTimeout

Reply:
if never emits event, worker never be ready to handle tasks. First we need discuss how timeout will work with 'autoStart' and 'lazyStart' options, and provide something like `maxWorkerRestarts`, to avoid infinite worker restarts loop.

for now, we can directly end process in worker if async initialization fails for some reason. For example:

```
// worker.js
const { sendWorkerMessage } = require('worker-nodes');

setup()
  .then(() => sendWorkerMessage('ready'))
  .catch(() => process.exit(-1))

module.exports = {
    someTask: () => {} 
};
```